pandas/_libs/groupby.pyx

```diff
@@ -263,6 +276,27 @@ def group_cumsum(numeric_t[:, ::1] out,
                         if not skipna:
                             accum[lab, j] = NaN
                             break
+                elif numeric_t == float64_t:
```

Comment:
I think you can avoid duplicating these blocks through https://cython.readthedocs.io/en/latest/src/userguide/fusedtypes.html#type-checking-specializations

eg something like (`if numeric_t == float32_t and (val == INF32 or val == -INF32)) (then same for float64) should create float32 and float64 specific conditions at compile time (so no runtime penalty)

Reply:
Also fused type checking may have to be with `is` instead of `==`, but not sure if that matters

Reply:
Thanks for the hint. `numeric_t` already is a fused type therefore I added a function which checks if its single argument is `inf` depending on the type.